### PR TITLE
feat(config): Deprecate `fieldpass` and `fielddrop` modifiers

### LINKED ISSUE #12675: Deprecate `fieldpass` and `fielddrop` modifiers
### Use Case

The current naming is not consistent and this it is not clear what it does. We have had several users on the community Slack/forums (including me) that were confused by this naming.

Let's take a look at current documentation: [Metric filtering](https://github.com/influxdata/telegraf/blob/v1.25.2/docs/CONFIGURATION.md#metric-filtering)

You have the following selectors which indicate wether or not a metric will be emitted for a particular plugin:

- name**pass**
- name**drop**
- tag**pass**
- tag**drop**

*Note that they all have the same consistent suffix of `pass` or `drop`.*

Then we have the following modifiers which indicate wether or not an item will be dropped from the metric:

- field**pass**
- field**drop**
- tag**include**
- tag**exclude**

So here, one would expect something like field-include/exclude as field-pass/drop may sound like they indicate the metric not being emitted to the plugin.

### Expected behavior

Being able to use `fieldinclude` and `fieldexclude`  for any plugin to be more consistent with the already existing `taginclude` and `tagexclude`.

When still using `fieldpass` and `fielddrop` a deprecation warning should be emitted at startup.

### Actual behavior

Only able to use `fieldpass` and `fielddrop`.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Hipska**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #12675

### CODE REVIEW

**Comment by Hipska** [comment]
!retry-failed

**Review by srebhan — COMMENTED** [review]
Just one request to add a deprecation warning if the old options are used...

File: config/config.go [hunk 1]
```diff
@@ -1434,9 +1434,11 @@ func (c *Config) buildFilter(tbl *ast.Table) (models.Filter, error) {
 
 	c.getFieldStringSlice(tbl, "pass", &f.FieldPass)
 	c.getFieldStringSlice(tbl, "fieldpass", &f.FieldPass)
+	c.getFieldStringSlice(tbl, "fieldinclude", &f.FieldPass)
 
 	c.getFieldStringSlice(tbl, "drop", &f.FieldDrop)
 	c.getFieldStringSlice(tbl, "fielddrop", &f.FieldDrop)
+	c.getFieldStringSlice(tbl, "fieldexclude", &f.FieldDrop)
```
[the remark this answers is not in the document]
Comment (Hipska) [reply config/config.go:1441]: Sure, was going to add that, this PR is not finished yet, it's still in Draft.

File: migrations/common/input_options.go [hunk 2]
```diff
@@ -13,15 +13,5 @@ type InputOptions struct {
...
-	FieldPass      []string            `toml:"fieldpass,omitempty"`
-	FieldDropOld   []string            `toml:"drop,omitempty"`
-	FieldDrop      []string            `toml:"fielddrop,omitempty"`
-	TagPassFilters map[string][]string `toml:"tagpass,omitempty"`
-	TagDropFilters map[string][]string `toml:"tagdrop,omitempty"`
-	TagExclude     []string            `toml:"tagexclude,omitempty"`
-	TagInclude     []string            `toml:"taginclude,omitempty"`
-	MetricPass     string              `toml:"metricpass,omitempty"`
+	FilterOptions
 }
```
Comment (srebhan) [inline migrations/common/input_options.go:17]: Have you tested this? This does not work when marshalling the data again!

File: migrations/common/output_options.go [hunk 3]
```diff
@@ -5,15 +5,5 @@ type OutputOptions struct {
...
-	FieldPassOld   []string            `toml:"pass,omitempty"`
-	FieldPass      []string            `toml:"fieldpass,omitempty"`
-	FieldDropOld   []string            `toml:"drop,omitempty"`
-	FieldDrop      []string            `toml:"fielddrop,omitempty"`
-	TagPassFilters map[string][]string `toml:"tagpass,omitempty"`
-	TagDropFilters map[string][]string `toml:"tagdrop,omitempty"`
-	TagExclude     []string            `toml:"tagexclude,omitempty"`
-	TagInclude     []string            `toml:"taginclude,omitempty"`
-	MetricPass     string              `toml:"metricpass,omitempty"`
+	FilterOptions
```
Comment (srebhan) [inline migrations/common/output_options.go:8]: See above.

**Comment by Hipska** [comment]
I think I should probably also rename these? https://github.com/influxdata/telegraf/blob/4c4b821aaa22cc48dc91412c10d2ef8e3fb9c7e3/models/filter.go#L41-L44

And this one should be removed, as it is simply untrue: https://github.com/influxdata/telegraf/blob/4c4b821aaa22cc48dc91412c10d2ef8e3fb9c7e3/models/filter.go#L199-L203

**Review by srebhan — COMMENTED** [review]
Thank you very much for your work @Hipska! IIRC the embedding of filter options will not work when marshalling the struct to TOML and thus we cannot do this.
Furthermore, it would be awesome to have a migration for the options, but that might be another PR...

File: migrations/common/input_options.go (hunk 2 above)
Comment (Hipska) [reply migrations/common/input_options.go:17]: The tests seem to succeed? Although it looks like only the one for riemann has an actual test with `namepass` and `fieldpass` in it, and that one also works fine..

Comment (srebhan) [reply migrations/common/input_options.go:17]: Would you care to add a testcase to `migrations/inputs_httpjson/testcases`? This migration uses the above structure IIRC.

Comment (Hipska) [reply migrations/common/input_options.go:17]: It looks like that is a limitation of the used TOML library (which is kinda outdated as well..)

File: models/filter.go [hunk 4]
```diff
@@ -208,11 +208,11 @@ func (f *Filter) shouldTagsPass(tags []*telegraf.Tag) bool {
 	return ShouldTagsPass(f.TagPassFilters, f.TagDropFilters, tags)
 }
 
-// filterFields removes fields according to fieldpass/fielddrop.
+// filterFields removes fields according to fieldinclude/fieldexclude.
 func (f *Filter) filterFields(metric telegraf.Metric) {
 	filterKeys := []string{}
 	for _, field := range metric.FieldList() {
-		if !f.shouldFieldPass(field.Key) {
+		if !ShouldPassFilters(f.fieldPassFilter, f.fieldDropFilter, field.Key) {
```
Comment (srebhan) [inline models/filter.go:215]: You then should also remove the replaced function (`f.shouldFieldPass`).

**Review by srebhan — COMMENTED** [review]
Two more minor comments...

**Review by srebhan — APPROVED** [review]
Thanks @Hipska! LGTM.

**Review by p-zak — CHANGES_REQUESTED** [review]
Small cosmetic thing from me :)

**Review by powersj — COMMENTED** [review]
If you can address p-zak's comment + resolve conflicts, we can land this. Thanks!

**Pull request merged**